### PR TITLE
Create backup collections for Group's default SharePoint site

### PR DESCRIPTION
**Pull request by meain**
This commit has the initial rough set of changes needed to create collections from the group's default SharePoint site.

This still does not have all the functionality that we need, but the idea was that we could get this in and iterate over time.

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [x] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* https://github.com/alcionai/corso/issues/3990

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by meain** [comment]
@ryanfkeepers I've rebased this to be on top your changes in https://github.com/alcionai/corso/pull/4057

File: src/pkg/path/category_type.go [hunk 1]
```diff
@@ -73,6 +73,9 @@ var serviceCategories = map[ServiceType]map[CategoryType]struct{}{
 		ListsCategory:     {},
 		PagesCategory:     {},
 	},
+	GroupsService: {
+		LibrariesCategory: {}, // TODO(meain)
```
Comment (ryanfkeepers) [inline src/pkg/path/category_type.go:77]: What's the TODO here mean?  There doesn't seem to be any additional implementation beyond what you already have.  Can you please spell out the expected work in the todo comment?

File: src/internal/m365/backup.go [hunk 2]
```diff
@@ -197,7 +214,8 @@ func checkServiceEnabled(
 	service path.ServiceType,
 	resource string,
 ) (bool, bool, error) {
-	if service == path.SharePointService {
+	if service == path.SharePointService || service == path.GroupsService {
+		// TODO(meain): use proper check for groups
```
Comment (ryanfkeepers) [inline src/internal/m365/backup.go:218]: What is the "proper" check, in this case?

**Review by ryanfkeepers — APPROVED** [review]
Generally looks good.  Things to fix up, but ready to unblock, thus approved.

Also, this PR has slightly stepped over the size line into being too big.  Enough so that I don't feel entirely confident that I've been able to catch all possible gotchas in this one review: too much cognitive load to juggle in one go.  In the future it would be helpful to have these changes chunked out into separate PRs, even if that means some of the early ones fail builds due to missing/stubbed dependencies and such.  It allows me, as a reviewer, to concentrate more completely on one topic at a time.

File: src/internal/m365/collection/drive/collections.go [hunk 3]
```diff
@@ -439,7 +439,7 @@ func (c *Collections) Get(
 	service, category := c.handler.ServiceCat()
 	md, err := graph.MakeMetadataCollection(
 		c.tenantID,
-		c.resourceOwner,
+		c.resourceOwner, // TODO(meain): path fixes: group id
```
[the remark this answers is not in the document]
Comment (meain) [reply src/internal/m365/collection/drive/collections.go:442]: This as of now shows up as sharepoint site id for group site backup. That said, this is not the right place for the TODO. I've moved it up the func call with a better message.

File: src/internal/m365/collection/drive/group_handler.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		odConsts.DrivesPathDir,
+		driveID,
+		odConsts.RootPathDir)
+}
+
+func (h groupBackupHandler) CanonicalPath(
+	folders *path.Builder,
+	tenantID, resourceOwner string,
+) (path.Path, error) {
+	// TODO(meain): resourceOwner
```
[the remark this answers is not in the document]
Comment (meain) [reply src/internal/m365/collection/drive/group_handler.go:56]: I've updated the todo to be more specific.

File: src/internal/m365/collection/drive/group_handler.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
+package drive
```
[the remark this answers is not in the document]
Comment (meain) [reply src/internal/m365/collection/drive/group_handler.go:1]: Yup, that makes sense. Let me merge them. I'll move that logic to a separate PR to make it easier to review though.

**Pull request merged**